src/bin/ion/commands/generate/templates/java/class.templ

```diff
@@ -39,6 +39,45 @@ import java.io.IOException;
         }
     {% endfor %}
 
+    public static class {{ model.name }}Builder {
+        {% for field_name, field_val in struct_info["fields"] -%}
+            {% set field_value = field_val.0 | fully_qualified_type_name %}
+            private java.util.Optional<{{ field_value | wrapper_class }}> {{ field_name | camel }};
+        {% endfor %}
+
+        {{ model.name }}Builder() {
+            {# Initializes all the fields of this class #}
```

Comment:
They can also be initialized at the declaration site, and they you don't need to implement any constructor at all. It can just use the default constructor.